functions/pagination-parameters.js

```diff
@@ -0,0 +1,200 @@
+// - if present, `skip` must be an integer, optional, with a default value of 0
+// - if present, `maxpagesize` must be an integer, optional, with no default value
+// - if present, `filter` must be a string and optional
+// - if present, `orderby` should be be an array of strings and optional
+// - if present, `select` should be be an array of strings and optional
+// - if present, `expand` should be be an array of strings and optional
+
+module.exports = (operation, _opts, paths) => {
+  // operation should be a get or post operation
+  if (operation === null || typeof operation !== 'object') {
+    return [];
+  }
+  const path = paths.path || paths.target || [];
+
+  // If the operation has no parameters, there is nothing to check
+  if (!operation.parameters) {
+    return [];
+  }
+
+  const errors = [];
+
+  // Check the top parameter
+  const topIndex = operation.parameters.findIndex((param) => param.name?.toLowerCase() === 'top');
+  if (topIndex !== -1) {
+    const top = operation.parameters[topIndex];
+    // Check that top is an integer
+    if (top.type !== 'integer') {
+      errors.push({
+        message: 'top parameter must be type: integer',
+        path: [...path, 'parameters', topIndex, 'type'],
+      });
+    }
+    // Check that top is optional
+    if (top.required) {
+      errors.push({
+        message: 'top parameter must be optional',
+        path: [...path, 'parameters', topIndex, 'required'],
+      });
+    }
+    // Check that top has no default value
+    if (top.default !== undefined) {
+      errors.push({
+        message: 'top parameter must have no default value',
+        path: [...path, 'parameters', topIndex, 'default'],
+      });
+    }
+  }
+
+  // Check skip parameter
+  const skipIndex = operation.parameters.findIndex((param) => param.name?.toLowerCase() === 'skip');
+  if (skipIndex !== -1) {
+    const skip = operation.parameters[skipIndex];
+    // Check that skip is an integer
+    if (skip.type !== 'integer') {
+      errors.push({
+        message: 'skip parameter must be type: integer',
+        path: [...path, 'parameters', skipIndex, 'type'],
+      });
+    }
+    // Check that skip is optional
+    if (skip.required) {
+      errors.push({
+        message: 'skip parameter must be optional',
+        path: [...path, 'parameters', skipIndex, 'required'],
+      });
+    }
+    // Check that skip has a default value of 0
+    if (skip.default !== 0) {
+      errors.push({
+        message: 'skip parameter must have a default value of 0',
+        path: [...path, 'parameters', skipIndex, 'default'],
+      });
+    }
+  }
+
+  // Check maxpagesize parameter
+  const maxpagesizeIndex = operation.parameters.findIndex((param) => param.name?.toLowerCase() === 'maxpagesize');
+  if (maxpagesizeIndex !== -1) {
+    const maxpagesize = operation.parameters[maxpagesizeIndex];
+    // Check case convention for maxpagesize
+    if (maxpagesize.name !== 'maxpagesize') {
```

Comment:
Perhaps we should check for casing for all of the params? In case it's pascal-cased?